analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,148 @@
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            base.Initialize(context);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                if (IsEnabled(compilationContext.Options))
+                {
+                    context.RegisterSyntaxNodeActionInNonGenerated(VisitAttribute, SyntaxKind.Attribute);
+                }
+            });
+        }
+
+        protected override void Initialize(TrackerInput input)
+        {
+            SetupInvocationTracker(Language.Tracker.Invocation, input);
+            SetupObjectCreationTracker(Language.Tracker.ObjectCreation, input);
+        }
+
+        private static void SetupInvocationTracker(InvocationTracker<SyntaxKind> tracker, TrackerInput input)
+        {
+            tracker.Track(input,
+                          tracker.MatchMethod(new MemberDescriptor(KnownType.System_Collections_Generic_IDictionary_TKey_TValue, "Add")),
+                          tracker.MethodHasParameters(ParameterCount),
+                          c => IsFirstArgumentAccessControlAllowOrigin((InvocationExpressionSyntax)c.Node, c.SemanticModel)
+                               && IsSecondArgumentStarString((InvocationExpressionSyntax)c.Node, c.SemanticModel),
+                          tracker.IsIHeadersDictionary());
+
+            tracker.Track(input,
+                          tracker.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Cors_Infrastructure_CorsPolicyBuilder, "WithOrigins")),
+                          c => ContainsStar(((InvocationExpressionSyntax)c.Node).ArgumentList.Arguments.Select(a => a.Expression), c.SemanticModel));
+
+            tracker.Track(input, tracker.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Cors_Infrastructure_CorsPolicyBuilder, "AllowAnyOrigin")));
+
+            tracker.Track(input,
+                          tracker.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Http_HeaderDictionaryExtensions, "Append"),
+                                              new MemberDescriptor(KnownType.System_Web_HttpResponse, "AppendHeader"),
+                                              new MemberDescriptor(KnownType.System_Web_HttpResponseBase, "AddHeader"),
+                                              new MemberDescriptor(KnownType.System_Collections_Specialized_NameValueCollection, "Add"),
+                                              new MemberDescriptor(KnownType.System_Net_Http_Headers_HttpHeaders, "Add")),
+                          tracker.MethodHasParameters(ParameterCount),
+                          c => IsFirstArgumentAccessControlAllowOrigin((InvocationExpressionSyntax)c.Node, c.SemanticModel)
+                               && IsSecondArgumentStarString((InvocationExpressionSyntax)c.Node, c.SemanticModel));
+        }
+
+        private static void SetupObjectCreationTracker(ObjectCreationTracker<SyntaxKind> tracker, TrackerInput input) =>
```

Comment:
this is going to be called for all object creations - can you do an early-return based on a simple string check inside the argument list, if it contains "*"?

because it's going to get the symbol for all object creations, and that's a lot of symbols to check

or maybe do the arguments check before?... that would work if inside `ContainsStar` the argumnets check would be done before the method symbol check

Reply:
I've added `ObjectCreationExpressionSyntaxExtensions.IsKnownType` which checks first the name of the type before retrieving the symbol.